File: practice/algorithms/implementation/lisas_workbook/LisasWorkbook.js
'use strict';

const processData = input => {
    const lines = input.split('\n');
    const arr = lines[0].split(/\s+/).map(i => parseInt(i));
    const n = arr[0];
    const k = arr[1];
    const problemsPerChapter = lines[1].split(/\s+/).map(i => parseInt(i));
    let specialProblems = 0;
    let page = 1;
    for(let i = 0; i < n; i++) {
        const problems = problemsPerChapter[i];
        for(let p = 1; p <= problems; p++) {
            if(p === page) {
                specialProblems++;
            }
            if(p !== problems && p % k === 0) {
                page++;
            }
        }
        page++;
    }
    console.log(specialProblems);
};

process.stdin.resume();
process.stdin.setEncoding("ascii");

var _input = "";
process.stdin.on("data", input => _input += input);
process.stdin.on("end", () => processData(_input));

File: practice/algorithms/implementation/lisas_workbook/lisas_workbook.rb
(n, k) = gets.split.map {|i| i.to_i}
problems_per_chapter = gets.split.map {|i| i.to_i}
page = 1
special_problems = 0
n.times do |i|
  problems = problems_per_chapter[i]
  for p in 1..problems
    special_problems += 1 if p == page
    page += 1 if p != problems && p % k == 0
  end
  page += 1
end

puts special_problems